pennylane/gradients/parameter_shift.py

```diff
@@ -140,34 +140,73 @@ def _extract_unshifted(recipe, at_least_one_unshifted, f0, gradient_tapes, tape)
     to evaluate a gradient result."""
-
     _, coeffs, fn, unshifted_coeff, batch_size = data
 
     # individual post-processing of e.g. Hamiltonian grad tapes
     if fn is not None:
         res = fn(res)
 
-    # compute the linear combination of results and coefficients
-    axis = 0
-    if not broadcast:
-        res = qml.math.stack(res)
-    elif (
-        qml.math.get_interface(res[0]) != "torch"
-        and batch_size is not None
-        and not scalar_qfunc_output
-    ):
-        # If the original output is not scalar and broadcasting is used, the second axis
-        # (index 1) needs to be contracted. For Torch, this is not true because the
-        # output of the broadcasted tape is flat due to the behaviour of the Torch device.
-        axis = 1
-    g = qml.math.tensordot(res, qml.math.convert_like(coeffs, res), [[axis], [0]])
+    if not qml.active_return():
 
-    if unshifted_coeff is not None:
-        # add the unshifted term
-        g = g + unshifted_coeff * r0
+        axis = 0
+        if not broadcast:
+            res = qml.math.stack(res)
+        elif (
+            qml.math.get_interface(res[0]) != "torch"
+            and batch_size is not None
+            and not scalar_qfunc_output
+        ):
+            # If the original output is not scalar and broadcasting is used, the second axis
+            # (index 1) needs to be contracted. For Torch, this is not true because the
+            # output of the broadcasted tape is flat due to the behaviour of the Torch device.
+            axis = 1
+
+        # compute the linear combination of results and coefficients
+        g = qml.math.tensordot(res, qml.math.convert_like(coeffs, res), [[axis], [0]])
+
+        g = _unshifted_coeff(g, unshifted_coeff, r0)
+
+    else:
+        multi_measure = len(tape.measurements) > 1
+        if not multi_measure:
+
+            res = qml.math.stack(res)
+
+            if (
+                tape.measurements[0].return_type == qml.measurements.Probability
+                or len(res.shape) > 1
+            ):
+                res = qml.math.squeeze(res)
+
+            g = qml.math.tensordot(res, qml.math.convert_like(coeffs, res), [[0], [0]])
+            g = _unshifted_coeff(g, unshifted_coeff, r0)
+        else:
+            # New return type output
+            g = []
+
+            # Multiple measurements case, so we can extract the first result
+            num_params = len(res[0])
+            for parameter_idx in range(num_params):
+
+                # Gather the measurement results
+                single_result = [measurement_result[parameter_idx] for measurement_result in res]
+                coeffs = qml.math.convert_like(coeffs, single_result)
+                g_component = qml.math.tensordot(single_result, coeffs, [[0], [0]])
+                g.append(g_component)
+
+            g = _unshifted_coeff(g, unshifted_coeff, r0)
+            g = tuple(g)

```

Comment:
> Just curious where the shot vector axis is supposed to come from? Is the gradient computation for each component of the shot vector separate?

That would be ideal, though it has not yet been tested here yet, so likely that needs another push. I'm hesitant to include it in this PR though because of the number of additions here already.

Reply:
Yes agree that shoot vectors will need another PR.